reporter/src/main/kotlin/reporters/NoticeReporter.kt

```diff
@@ -173,12 +170,16 @@ class NoticeReporter : Reporter() {
                     null
                 }
             }.joinToString(separator = NOTICE_SEPARATOR, prefix = NOTICE_SEPARATOR)
-            outputFile.appendText(findings)
+        } else {
+            ""
         }
 
-        if (noticeReport.footers.isNotEmpty()) {
-            val footers = noticeReport.footers.joinToString(separator = NOTICE_SEPARATOR, prefix = NOTICE_SEPARATOR)
-            outputFile.appendText(footers)
+        val footers = if (noticeReport.footers.isNotEmpty()) {
+            noticeReport.footers.joinToString(separator = NOTICE_SEPARATOR, prefix = NOTICE_SEPARATOR)
+        } else {
+            ""
         }
+
+        return headers + findings + footers
```

Comment:
You could get rid of the `else { "" }` blocks by using:
```kotlin
private fun writeNoticeReport(noticeReport: NoticeReport) =
  buildString {
    ...
  }
```

Reply:
Good idea, done.